docs/fundamentals/code-analysis/quality-rules/ca2259.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "CA2259: Ensure 'ThreadStatic' is only used with static fields"
+description: "Learn about code analysis rule CA2259: Ensure 'ThreadStatic' is only used with static fields"
+ms.date: 11/04/2022
+ms.topic: reference
+f1_keywords:
+- CA2259
+helpviewer_keywords:
+- CA2259
+dev_langs:
+- CSharp
+- VB
+---
+# CA2259: Ensure `ThreadStatic` is only used with static fields
+
+|                                     | Value                                |
+| ----------------------------------- | ------------------------------------ |
+| **Rule ID**                         | CA2259                               |
+| **Category**                        | [Usage](usage-warnings.md)           |
+| **Fix is breaking or non-breaking** | Non-breaking                         |
+
+## Cause
+
+The <xref:System.ThreadStaticAttribute> attribute is applied to an instance field.
+
+## Rule description
+
+<xref:System.ThreadStaticAttribute>, which indicates that the value of a field is unique for each thread, only affects `static` (`Shared` in Visual Basic) fields. When applied to instance fields, the attribute has no impact on behavior.
+
+## How to fix a violation
+
+To fix a violation, remove the <xref:System.ThreadStaticAttribute> attribute from the field.
+
+## Example
+
+The following code snippet shows a violation of CA2259:
+
+```csharp
+class C
+{
+    [ThreadStatic]
+    public int number = 404;
+}
+```
+
+```vb
+Class C
+    <ThreadStatic>
+    Public number As Integer = 404
+End Class
+```
+
+## When to suppress warnings
+
+It's safe to suppress a warning from this rule.
```

Comment:
You can suppress it, but the [ThreadStatic] is a nop and isn't doing what you intend it to do.